corehq/apps/domain/extension_points.py

```diff
@@ -13,3 +13,8 @@ def custom_domain_module(domain) -> Optional[str]:
     Returns:
         A string of the python path to the module for the domain, or None
     """
+
+
+@extension_point(result_format=ResultFormat.FIRST)
+def custom_clean_password(password) -> Optional[tuple]:
+    """Custom function to ensure that password meets requirements"""
```

Comment:
Would be good to include docs for the what's expected from the return value.
```suggestion
    """Custom function to ensure that password meets requirements
    
    Returns:
        A tuple with two items:
            1. strength (integer): a value less than 2 is considered weak
            2. message (string): message text to be shown if the password is not strong enough
    """
```

Reply:
Thinking about this some more I would change the way this works to just return an error message or None:
```
message = check_password_strength()
if message:
    raise forms.ValidationError(message)
```

Returning the strength is a bit pointless since we're not actually calculating a strength in all cases, we're just doing a binary check.

Reply:
That makes perfect sense. I will update.